core/pkg/launch/wandb_config.go

```diff
@@ -0,0 +1,80 @@
+package launch
+
+import (
+	"github.com/wandb/wandb/core/internal/data_types"
+	"github.com/wandb/wandb/core/pkg/service"
+)
+
+// Selector for job inputs from the wandb.config.
+//
+// The includePaths and excludePaths are used to filter down the run config
+// before it is converted to a schema and saved as job inputs.
+type launchWandbConfigParameters struct {
+	includePaths []ConfigPath
+	excludePaths []ConfigPath
+}
+
+func newWandbConfigParameters() *launchWandbConfigParameters {
+	return &launchWandbConfigParameters{[]ConfigPath{}, []ConfigPath{}}
+}
+
+func (p *launchWandbConfigParameters) appendIncludePaths(
+	includePaths []*service.ConfigFilterPath,
+) {
+	for _, path := range includePaths {
+		p.includePaths = append(p.includePaths, path.Path)
+	}
+}
+
+func (p *launchWandbConfigParameters) appendExcludePaths(
+	excludePaths []*service.ConfigFilterPath,
+) {
+	for _, path := range excludePaths {
+		p.excludePaths = append(p.excludePaths, path.Path)
+	}
+}
+
+func (p *launchWandbConfigParameters) include() []ConfigPath {
+	return p.includePaths
+}
+
+func (p *launchWandbConfigParameters) exclude() []ConfigPath {
+	return p.excludePaths
+}
+
+// Create a typed representation of the wandb config inputs.
+//
+// Loads config filters received by the internal process, uses them to filter
+// down a copy of the run config, and then produces a TypedRepresentation for
+// the filtered config.
+//
+// If there are any errors in the process, the function logs them and returns
+// an unknown type representation. The errors should never happen in practice.
```

Comment:
Error handling should usually be pushed as high up as possible. In this case it would be better to let the calling code log the error and explicitly use an unknown type representation.

Otherwise, the calling code looks like it has no edge cases!